libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -176,6 +177,24 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 	yulAssert(m_scope, "");
 
 	size_t const numVariables = _varDecl.variables.size();
+
+	// Special case of ``let x := y`` where ``y`` is part of the current scope and this is the last reference to ``y``.
+	// In this case just delete y and assign x y's stack slot.
+	if (Identifier* rhsIdentifier = std::get_if<Identifier>(_varDecl.value.get()))
+		if (Scope::Variable* rhs = std::get_if<Scope::Variable>(m_scope->lookup(rhsIdentifier->name, false)))
```

Comment:
Where is the "same scope" check done, here?

Reply:
That's the ``false`` - I added ``_superScopes`` as optional argument to ``lookup``.

Reply:
Ah right!